frontend/src/components/header/MenuOverlay.js

```diff
@@ -96,18 +97,23 @@ class MenuOverlay extends Component {
     if (targetValue) {
       this.setState({
         query: e.target.value,
+        pendingQuery: e.target.value ? true : false,
       });
 
       queryPathsRequest(e.target.value, 9)
         .then((response) => {
-          this.setState({
-            queriedResults: flattenLastElem(response.data),
-          });
+          if (e.target.value === this.queryString) {
+            this.setState({
+              queriedResults: flattenLastElem(response.data),
+              pendingQuery: false,
```

Comment:
set also `query: query`

Reply:
to know which is the query that was executed in order to get those `queriedResults`

Reply:
as I said, set the query too... because we are running async and we want to be consistent/on the safe side.

Reply:
done